nixos/modules/flyingcircus/roles/loghost.nix

```diff
@@ -108,7 +108,34 @@ in
     (mkIf cfg.enable {
 
       # XXX Access should *only* be allowed from directory and same-rg.
-      networking.firewall.allowedTCPPorts = [ port ];
+      networking.firewall.allowedTCPPorts = [ 9000 ];
+
+      flyingcircus.roles.nginx.enable = true;
+      flyingcircus.roles.nginx.httpConfig =
+      let
+          listenOn = lib.concatMapStringsSep "\n    "
+              (addr: "listen ${addr}:9000;")
+              (fclib.listenAddressesQuotedV6 config "ethsrv");
+
+          allow = lib.concatMapStringsSep "\n    "
+              (addr: "allow ${addr};")
+              (config.flyingcircus.static.directory.proxy_ips);
+
+
+      in
+      ''
+        server {
+            ${listenOn}
+
+            ${allow}
+            deny all;
+
+            location /tools/${config.flyingcircus.enc.name}/graylog {
+                proxy_pass http://localhost:9001;
```

Comment:
`localhost` resolves to `::1` first. If the connection attempt fails, try `127.0.0.1` in a second attempt. It would be beneficial to use either the numeric variant in all places or listen consequently on `::1` to avoid unnecessary connect overheads.